public/src/managers/ServerProvider.ts

```diff
@@ -243,6 +243,67 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
         return JSON.parse(JSON.stringify(result.statusCode)) as IStatusCode;
     }
 
+    // getAllGroupLabInfos 
+    public async getAllGroupLabInfos(courseId: number, groupID: number): Promise<IMap<ISubmission>> { 
+        const result = await this.helper.get<ISubmission[]>(
+            ("courses/" + courseId.toString() + "/groups/" + groupID + "/submissions"),
+        );
+        if (result.statusCode === 200 && result.data === undefined) {
+            return {};
+        }
+        if (!result.data) {
+            this.handleError(result, "getAllLabInfos");
+            return {};
+        }
+        return mapify(result.data, (e) => {
+            let a = "{\"builddate\": \"2017-07-28\", \"buildid\": 1, \"buildlog\": \"This is cool\", \"execTime\": 1}";
```

Comment:
But then again, looking onwards in the code, I'm not quite sure why these `a` and `b` variables needs to be initialized like this. I think they could be just empty strings as well, since they will be overwritten in all cases.

Reply:
Also, since `a` and `b` are used several times below and somewhat far apart from this definition, they should probably have more descriptive names. General rule: If you are only using the variables very close to where they are defined, it is sometimes ok to use single letter variable names. The typical example is loop index variables such as `i` and `j`, as long as they are used close to where they are defined.

Reply:
I tested removing the initialized text for both strings a and b, seems like it would work.
The problem removing the text is if any of the two following if statements fail, the corresponding output would be empty when shown to the user.  But this is taken care of by the try/catch on the parsing, since parsing a empty string raises an error.

So i'll remove the initialized strings and rename them